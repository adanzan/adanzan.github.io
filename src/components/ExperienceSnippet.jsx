/* eslint no-unused-vars: 0 */

/**
 * Features the details of an organization, along with the roles I worked as
*/

import * as React from "react";
import PropTypes from "prop-types";
import Paper from "@mui/material/Paper";

import styles from "../styles/Snippet.module.css";

export default function ExperienceSnippet({
	name, location, logo, link, positions
}) {
	// Creates an element for each position in the organization
	const positionElement = positions.map((position) => (
		<div className={styles.position} key={position.name}>
			<p>
				{" "}
				<strong>{position.name}</strong>
			</p>
			<p>{position.dateRange}</p>
		</div>
	));

	return (
		<div className={styles.ExperienceSnippet}>
			<Paper className={styles.snippetPaper} elevation={0}
				sx={{
					border: "1.5px solid orange", 
					borderRadius: "1em"
				}}>
				
				<div className={styles.org}>
					<a href={link} target="_blank" rel="noopener noreferrer">
						<img className={styles.logo} src={logo} alt={`${name} logo`} />
					</a>
					<div className={styles.nameLocationWrapper}>
						<div className={styles.nameLocation}>
							<a className={styles.name} href={link} target="_blank" rel="noopener noreferrer">
								<h3>{name}</h3>
							</a>
							<p className={styles.location}>{location}</p>
						</div>
					</div>
				</div>
				<div className={styles.detailsWrapper}>
					{positionElement}
				</div>
			</Paper>
		</div>
	);
}

ExperienceSnippet.propTypes = {
	name: PropTypes.string.isRequired,
	location: PropTypes.string.isRequired,
	logo: PropTypes.string.isRequired,
	link: PropTypes.string.isRequired,
	positions: PropTypes.arrayOf(
		PropTypes.shape({
			name: PropTypes.string.isRequired,
			dateRange: PropTypes.string.isRequired,
		}),
	).isRequired,
};
