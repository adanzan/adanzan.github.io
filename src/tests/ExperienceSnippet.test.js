/* eslint no-unused-vars: 0 */
import * as React from "react";
import { render } from "@testing-library/react";

import ExperienceSnippet from "../components/ExperienceSnippet"; 

let testRender;

describe("ExperienceSnippet component", () => {
	beforeEach(() => {
		testRender = render(<ExperienceSnippet 
			name={"testName"} 
			location={"testLocation"} 
			logo={"testLogo"} 
			link={"testLink"} 
			positions={[{name: "testName1", dateRange: "testDateRange1"}, 
				{name: "testName2", dateRange: "testDateRange2"}]}
		/>);
	});
	test("matches snapshot", () => {
		expect(testRender).toMatchSnapshot();
	});
	describe("correctly displays", () => {
		test("name", () => {
			expect(testRender.getByRole("heading", "name")).toContainHTML("testName");
		});
		test("positions", () => {
			expect(testRender.getByText("testName1"));
			expect(testRender.getByText("testDateRange2"));
		});
	});
});
		
		
