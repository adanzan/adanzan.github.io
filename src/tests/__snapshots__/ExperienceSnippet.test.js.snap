// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceSnippet component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ExperienceSnippet"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 snippetPaper css-14epxft-MuiPaper-root"
        >
          <div
            class="org"
          >
            <a
              href="testLink"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="testName logo"
                class="logo"
                src="testLogo"
              />
            </a>
            <div
              class="nameLocationWrapper"
            >
              <div
                class="nameLocation"
              >
                <a
                  class="name"
                  href="testLink"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <h3>
                    testName
                  </h3>
                </a>
                <p
                  class="location"
                >
                  testLocation
                </p>
              </div>
            </div>
          </div>
          <div
            class="detailsWrapper"
          >
            <div
              class="position"
            >
              <p>
                 
                <strong>
                  testName1
                </strong>
              </p>
              <p>
                testDateRange1
              </p>
            </div>
            <div
              class="position"
            >
              <p>
                 
                <strong>
                  testName2
                </strong>
              </p>
              <p>
                testDateRange2
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ExperienceSnippet"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 snippetPaper css-14epxft-MuiPaper-root"
      >
        <div
          class="org"
        >
          <a
            href="testLink"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="testName logo"
              class="logo"
              src="testLogo"
            />
          </a>
          <div
            class="nameLocationWrapper"
          >
            <div
              class="nameLocation"
            >
              <a
                class="name"
                href="testLink"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h3>
                  testName
                </h3>
              </a>
              <p
                class="location"
              >
                testLocation
              </p>
            </div>
          </div>
        </div>
        <div
          class="detailsWrapper"
        >
          <div
            class="position"
          >
            <p>
               
              <strong>
                testName1
              </strong>
            </p>
            <p>
              testDateRange1
            </p>
          </div>
          <div
            class="position"
          >
            <p>
               
              <strong>
                testName2
              </strong>
            </p>
            <p>
              testDateRange2
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
